### PR TITLE
fix: Promotional code/ Cards displayed according to ticket type

### LINKED ISSUE #5914: Ticket: If an event has only free and donation tickets do not show "promotional code" and other details
Please change the public ticket info as follows:
1. If an event has only free and donation tickets do not show "promotional code". (Also reduce the spacing between "Total" and "Order now" then)
2. If an event has only free tickets and no other tickets do not show the credit card icons neither.
3. Also ensure this works on mobile views.

Screenshot from 2020-12-05 08-38-48

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #5914

#### Short description of what this resolves:
If an event has only free and donation tickets do not show "promotional code". (Also reduce the spacing between "Total" and "Order now" then)
If an event has only free tickets and no other tickets do not show the credit card icons neither.

SCREENSHOTS -

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by daretobedifferent18 — DISMISSED** [review]
Great @MeghalBisht

File: app/components/public/ticket-list.js [hunk 1]
```diff
@@ -71,6 +71,14 @@ export default Component.extend(FormMixin, {
     ) > 0;
   }),
 
+  hasPaidTickets: computed('tickets.@each.type', function() {
+    return this.tickets.toArray().filter(ticket => ticket.type === 'paid').length > 0;
```
Comment (iamareebjamal) [inline app/components/public/ticket-list.js:75]: Donation tickets are paid as well
Just return !hasOnlyFreeTickets

Comment (MeghalBisht) [reply app/components/public/ticket-list.js]: Done!

File: app/templates/components/public/ticket-list.hbs [hunk 2]
```diff
@@ -168,7 +168,7 @@
         </div>
       </div>
     {{else}}
-      {{#if this.hasPaidTickets}}
+      {{#if (not this.hasOnlyFreeTickets)}}
```
Comment (iamareebjamal) [inline app/templates/components/public/ticket-list.hbs:171]: unless

**Comment by mariobehling** [comment]
Very good. Thank you!

**Comment by inayahassan23** [comment]
i was working on this i commented also why are u sending pull requst for every issue?? first try to solve ur already opened prs plz

**Comment by iamareebjamal** [comment]
@inayahassan23 There was no update for 4 days. Only comment when you have time to work on the issue

**Comment by inayahassan23** [comment]
ya sir but i was setting my code in my system so took some time and i didnt comment on any other issue..she already has so pending prs but in instructions it is mentioned to send only 1 pr

**Comment by iamareebjamal** [comment]
All her PRs are complete. Nothing stops anyone to take another issue once they are done on it before it is merged. Instructions also say only comment once you are ready to solve the issue. We can't reserve an issue for you for weeks. A lot of people comment that they are working on an issue and then nothing comes out of it. The goal here is to get as much bugs fixed as possible. There are already 166 issues open, a lot to choose from. I'm sure your efforts weren't wasted as you didn't start working on it. Now you can focus on others

**Pull request merged**